### PR TITLE
Add order filter as an employee

### PR DESCRIPTION
**Pull request by mdobosz3**
Employee can filter his orders: delivered, undelivered and all

### CODE REVIEW

File: spec/system/order_filter_spec.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  before do
+    driven_by(:rack_test)
+  end
+
+  context 'when an employee filters orders' do
+    let!(:employee) { create(:employee) }
+    let!(:delivered_order) { create(:order, status: 'delivered') }
+    let!(:not_delivered_order) { create(:order, status: 'not_delivered') }
+
+    it 'Filtering by delivered, not_delivered and all' do
```
Comment (ciejj) [inline spec/system/order_filter_spec.rb:15]: suggestion only - check out betterspec it's a convention of naming the specs

```suggestion
    it 'filters by the delivery status' do
```

**Review by ciejj — COMMENTED** [review]
Looks good, but we should take into consideration param value validation :)

File: spec/system/order_filter_spec.rb (hunk 1 above)
Comment (mdobosz3) [reply spec/system/order_filter_spec.rb]: Of course I will read about it

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -4,7 +4,11 @@ class OrdersController < ApplicationController
   before_action :authenticate_employee!
 
   def index
-    render :index, locals: { orders: Order.where(employee: current_employee) }
+    if params[:status].present?
+      render :index, locals: { orders: Order.filter_by_status(params[:status]) }
```
[the remark this answers is not in the document]
Comment (mdobosz3) [reply app/controllers/orders_controller.rb:8]: Naturally there will be an error :)
I added a condition that checks if my status is valid.

**Review by ciejj — APPROVED** [review]
Nice! 👌

**Review by oskarlakner — APPROVED** [review]

**Pull request merged**